lib/Raven/Middleware/ContextInterfaceMiddleware.php

```diff
@@ -24,18 +24,46 @@
 final class ContextInterfaceMiddleware
 {
     /**
-     * @var Context The context storage
+     * @var Context The user context
      */
-    private $context;
+    private $userContext;
+
+    /**
+     * @var Context The tags context
+     */
+    private $tagsContext;
+
+    /**
+     * @var Context The extra context
+     */
+    private $extraContext;
+
+    /**
+     * @var Context The runtime context
+     */
+    private $runtimeContext;
+
+    /**
+     * @var Context The server OS context
+     */
+    private $serverOsContext;
 
     /**
      * Constructor.
      *
-     * @param Context $context The context storage
+     * @param Context $userContext     The user context
+     * @param Context $tagsContext     The tags context
+     * @param Context $extraContext    The extra context
+     * @param Context $runtimeContext  The runtime context
+     * @param Context $serverOsContext The server OS context
      */
-    public function __construct(Context $context)
+    public function __construct(Context $userContext, Context $tagsContext, Context $extraContext, Context $runtimeContext, Context $serverOsContext)
```

Comment:
I don't like this very much. It would be frighteningly easy to mess up the arguments. Is there any way to fix this?

Reply:
I've changed the middleware to only work on a single context so that we will have as many middleware instances added to the stack as the number of contexts to be processed. This simolify a bit the `ContextInterfaceMiddleware` class

Reply:
I've changed the middleware to only work on a single context so that we will have as many middleware instances added to the stack as the number of contexts to be processed. This simplify a bit the `ContextInterfaceMiddleware` class